### PR TITLE
Filebeat mqtt add cleansession

### PR DESCRIPTION
**Pull request by dramis**
## What does this PR do?

Add clean_session parameter in config file of filebeat mqtt input plugin.

When clean_session is true, the state information is discarded at connect and disconnect. Setting cleansession to false keeps the state information. Default value true

## Why is it important?

With clean_session to false, it's possible to keep unread message to Mqtt server.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: filebeat/input/mqtt/config.go [hunk 1]
```diff
@@ -31,6 +31,7 @@ type mqttInputConfig struct {
 	ClientID string `config:"client_id" validate:"nonzero"`
 	Username string `config:"username"`
 	Password string `config:"password"`
+	CleanSession bool `config:"clean_session"`
```
[the remark this answers is not in the document]
Comment (dramis) [reply filebeat/input/mqtt/config.go]: Done.

**Comment by ycombinator** [comment]
Also, please remember to document the new setting in https://github.com/elastic/beats/blob/main/filebeat/docs/inputs/input-mqtt.asciidoc and add an entry in https://github.com/elastic/beats/blob/main/CHANGELOG.next.asciidoc.

File: CHANGELOG.next.asciidoc [hunk 2]
```diff
@@ -342,6 +342,8 @@ automatic splitting at root level, if root level element is an array. {pull}3415
 - Add device support for Azure AD entity analytics. {pull}35807[35807]
 - Improve CEL input performance. {pull}35915[35915]
 - Adding filename details from zip to response for httpjson {issue}33952[33952] {pull}34044[34044]
+- Add clean_session parameter to MQTT input .  {pull}35806[16204]
```
Comment (ycombinator) [inline CHANGELOG.next.asciidoc:345]:
```suggestion
- Add `clean_session` configuration setting for MQTT input.  {pull}35806[16204]
```

File: filebeat/docs/inputs/input-mqtt.asciidoc [hunk 3]
```diff
@@ -68,6 +68,17 @@ A client username used for authentication provided on the application level by t
 
 A client password used for authentication provided on the application level by the MQTT protocol.
 
+===== `clean_session`
+
+The clean_session flag indicates whether the client wants to establish a persistent session with the broker. 
+The default is `false`.
```
Comment (ycombinator) [inline filebeat/docs/inputs/input-mqtt.asciidoc:74]: I thought the default was `true`?

File: filebeat/docs/inputs/input-mqtt.asciidoc [hunk 4]
```diff
@@ -68,6 +68,17 @@ A client username used for authentication provided on the application level by t
...
+
+When clean_session is set to false, considered a persistent session, the broker stores all subscriptions for 
+the client and all missed messages for the client that subscribed with a Quality of Service (QoS) level 1 or 2. 
+
+In contrast, when CleanSession is set to true, the broker doesn’t retain any information for the client
```
Comment (ycombinator) [inline filebeat/docs/inputs/input-mqtt.asciidoc:79]: Consistency.

```suggestion
In contrast, when `clean_session` is set to true, the broker doesn’t retain any information for the client 
```

File: filebeat/docs/inputs/input-mqtt.asciidoc [hunk 5]
```diff
@@ -68,6 +68,17 @@ A client username used for authentication provided on the application level by t
 
 A client password used for authentication provided on the application level by the MQTT protocol.
 
+===== `clean_session`
+
+The clean_session flag indicates whether the client wants to establish a persistent session with the broker. 
+The default is `false`.
+
+When clean_session is set to false, considered a persistent session, the broker stores all subscriptions for
```
Comment (ycombinator) [inline filebeat/docs/inputs/input-mqtt.asciidoc:76]:
```suggestion
When `clean_session` is set to false, the session is considered to be persistent. The broker stores all subscriptions for 
```

File: filebeat/docs/inputs/input-mqtt.asciidoc (hunk 3 above)
Comment (dramis) [reply filebeat/docs/inputs/input-mqtt.asciidoc]: Your write, sorry for this mistake.

File: filebeat/docs/inputs/input-mqtt.asciidoc [hunk 6]
```diff
@@ -68,6 +68,17 @@ A client username used for authentication provided on the application level by t
 
 A client password used for authentication provided on the application level by the MQTT protocol.
 
+===== `clean_session`
+
+The clean_session flag indicates whether the client wants to establish a persistent session with the broker.
```
Comment (ycombinator) [inline filebeat/docs/inputs/input-mqtt.asciidoc:73]: Sorry, missed a spot in the review.

```suggestion
The `clean_session` flag indicates whether the client wants to establish a persistent session with the broker. 
```

**Comment by ycombinator** [comment]
@dramis CI is failing due to formatting errors.  Could you run `mage fmt` and commit the changes?  I ran it locally and it looks like the formatting of `filebeat/input/mqtt/config.go` changed.  I'm running Go version 1.19.10, as per what's in the [`.go-version` file](https://github.com/elastic/beats/blob/main/.go-version).

**Review by ycombinator — APPROVED** [review]
LGTM.  Thanks for the contribution, @dramis!

**Pull request merged**